Command/ResolveCacheCommand.php

```diff
@@ -21,68 +22,109 @@
 
 class ResolveCacheCommand extends Command
 {
-    /* @var FilterManager $filterManager */
-    private $filterManager;
+    use CacheCommandTrait;
 
-    /* @var FilterService $filterService */
-    private $filterService;
+    /**
+     * @var DataManager
+     */
+    private $dataManager;
 
-    public function __construct(FilterManager $filterManager, FilterService $filterService)
+    public function __construct(DataManager $dataManager, CacheManager $cacheManager, FilterManager $filterManager)
     {
-        $this->filterManager = $filterManager;
-        $this->filterService = $filterService;
+        parent::__construct();
 
-        parent::__construct('liip:imagine:cache:resolve');
+        $this->dataManager = $dataManager;
+        $this->cacheManager = $cacheManager;
+        $this->filterManager = $filterManager;
     }
 
-    protected function configure()
+    protected function configure(): void
     {
         $this
-            ->setDescription('Resolve cache for given path and set of filters.')
-            ->addArgument('paths', InputArgument::REQUIRED | InputArgument::IS_ARRAY, 'Image paths')
-            ->addOption(
-                'filters',
-                'f',
-                InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY,
-                'Filters list'
-            )->setHelp(<<<'EOF'
-The <info>%command.name%</info> command resolves cache by specified parameters.
-It returns list of urls.
-
-<info>php app/console %command.name% path1 path2 --filters=thumb1</info>
-Cache for this two paths will be resolved with passed filter.
-As a result you will get<info>
-    http://localhost/media/cache/thumb1/path1
-    http://localhost/media/cache/thumb1/path2</info>
-
-You can pass few filters:
-<info>php app/console %command.name% path1 --filters=thumb1 --filters=thumb2</info>
-As a result you will get<info>
-    http://localhost/media/cache/thumb1/path1
-    http://localhost/media/cache/thumb2/path1</info>
-
-If you omit --filters parameter then to resolve given paths will be used all configured and available filters in application:
-<info>php app/console %command.name% path1</info>
-As a result you will get<info>
-    http://localhost/media/cache/thumb1/path1
-    http://localhost/media/cache/thumb2/path1</info>
+            ->setName('liip:imagine:cache:resolve')
+            ->setAliases(['imagine:get'])
+            ->setDescription('Resolve cache entries for the given paths and filters.')
+            ->addArgument('path', InputArgument::REQUIRED | InputArgument::IS_ARRAY,
+                'Image file path(s) to run resolution on.')
+            ->addOption('filter', 'f', InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
+                'Filter(s) to use for image resolution; if none explicitly passed, use all filters.')
+            ->addOption('force', 'F', InputOption::VALUE_NONE,
+                'Force re-resolution of image, regardless of whether it has been previously cached.')
```

Comment:
does this mean the image in the cache will be re-generated? and does that mean this command not only shows what cached images exist, but does a cache warmup?

Reply:
mmmm yes, but it looks like it does the [same thing than before (it stores the image if it does not exist)](https://github.com/liip/LiipImagineBundle/blob/master/Service/FilterService.php#L82), but with this option it doesn't check if it's already stored

Reply:
that makes sense, given the name of this flag. can we tweak the description of the command and of this parameter to be a bit more explicit what this thing is doing?

Reply:
we could formulate this as "Force generating the image and writing the cache, regardless of whether a cached version already exists."

Reply:
perfect, thanks!